### ISSUE TITLE
purefa_hg incorrect check mode

### ISSUE THREAD

**Issue by sanderdescamps** [Pure-Storage-Ansible/FlashArray-Collection#289]
**Describe the bug**
The `purefa_hg` module always registers a change. Even when there is no change. 

**To Reproduce**
```
- name: Create hostgroups and add hosts and volumes
  purefa_hg:
    fa_url: "{{ inventory_hostname }}"
    api_token: "{{ api_keys[inventory_hostname] }}"
    hostgroup: "{{ item.host_group_name }}"
    host: "{{ item.hosts }}"
    volume: "{{ item.volumes }}"
  loop: "{{ host_group_list }}"
  delegate_to: localhost
```

**Expected behavior**
Even when all parameters of the module stay the same, the module outputs `changed=True`.

**Additional context**
I looked into the code and think I found the issue. I will create a pull request with the fix after I opened this issue.

**Comment by sanderdescamps** [comment]
When will these changes be published on Ansible Galaxy?

**Comment by sdodsley** [comment]
@sanderdescamps this fix should be in release 1.13.0 which is now available through Galaxy
